sensor/common/clusterentities/store.go

```diff
@@ -132,14 +163,15 @@ func (ed *EntityData) AddContainerID(containerID string, container ContainerMeta
 
 func (e *Store) updateMetrics() {
 	metrics.UpdateNumberContainersInEntityStored(len(e.containerIDMap))
+	metrics.UpdateNumberHistoricalContainersInEntityStored(len(e.historicalContainerIDs))
 }
 
-// Cleanup deletes all entries from store
+// Cleanup deletes all entries from store (or marks them as historical)
 func (e *Store) Cleanup() {
 	e.mutex.Lock()
 	defer e.mutex.Unlock()
 	defer e.updateMetrics()
-	e.initMaps()
+	e.resetMaps()
```

Comment:
I think the `historyMutex` should be lock before `e.updateMetrics` since we are reading the historical map there. Maybe it would be good to rename `resetMaps` to `resetMapsNoLock` and have both locks in the `Cleanup` function.

Reply:
As discussed in Slack, I will drop the history mutex completely as it is not needed.

Reply:
Done in [e55f8aa](https://github.com/stackrox/stackrox/pull/13155/commits/e55f8aa8c280b8a0d257cf5616cca41a99b97128)